utils/lister.cc

```diff
@@ -46,11 +50,7 @@ future<directory_entry> lister::guarantee_type(directory_entry de) {
         return make_ready_future<directory_entry>(std::move(de));
     } else {
         auto f = file_type((_dir / de.name.c_str()).native(), follow_symlink::no);
-        return f.then([dir = _dir, de = std::move(de)] (std::optional<directory_entry_type> t) mutable {
-            // If some FS error occurs - return an exceptional future
-            if (!t) {
-                return make_exception_future<directory_entry>(std::runtime_error(fmt::format("Failed to get {} type.", (dir / de.name.c_str()).native())));
-            }
+        return f.then([de = std::move(de)] (std::optional<directory_entry_type> t) mutable {
```

Comment:
But this changes the semantic of the `garantee_type`. It no longer guarantees anything. And the documentation of `garantee_type` needs to be updated.

Reply:
Updated. Thanks!